packages/lib/src/components/Card/Card.tsx

```diff
@@ -162,6 +165,53 @@ export class CardElement extends UIElement<CardElementProps> {
         }
     }
 
+    private fieldTypeToSnakeCase(fieldType) {
+        let str = camelCaseToSnakeCase(fieldType);
+        // SFs need their fieldType mapped to what the endpoint expects
+        if (ALL_SECURED_FIELDS.includes(fieldType)) {
+            str = str.substring(ENCRYPTED.length + 1); // strip 'encrypted_' off the string
+        }
+        return str;
+    }
+
+    private onFocus = obj => {
+        // console.log('### Card::onFocus:: fieldType', obj.fieldType, 'target', this.fieldTypeToSnakeCase(obj.fieldType));
+        this.submitAnalytics({
+            type: ANALYTICS_FOCUS_STR,
+            target: this.fieldTypeToSnakeCase(obj.fieldType)
+        });
+
+        // Call merchant defined callback
+        if (ALL_SECURED_FIELDS.includes(obj.fieldType)) {
+            this.props.onFocus?.(obj.event);
+        } else {
+            this.props.onFocus?.(obj);
+        }
+    };
+
+    private onBlur = obj => {
+        this.submitAnalytics({
+            type: ANALYTICS_UNFOCUS_STR,
+            target: this.fieldTypeToSnakeCase(obj.fieldType)
+        });
+
+        // Call merchant defined callback
+        if (ALL_SECURED_FIELDS.includes(obj.fieldType)) {
+            this.props.onBlur?.(obj.event);
+        } else {
+            this.props.onBlur?.(obj);
+        }
+    };
+
+    private onErrorAnalytics = obj => {
```

Comment:
Is it possible to add proper typescript types to the functions signatures and objects? 
We are doing big improvements on v6 regarding this, so it would be nice to try keep improving the types of the codebase.

Reply:
Will do